### PR TITLE
[RFC][UN-11823] Improve calendars and dropdowns in mobile

### PR DESCRIPTION
**Pull request by mmelo**
## JIRA Ticket

https://uniplaces.atlassian.net/browse/UN-11823

## Context

**TLDR: add on-demand native dropdown behaviour to uni-dropdown**

**Part 1 / 2** 
Implement behaviour on ember-cli-uniq to be used on SPA-Offer-Page's Full-edit mode Photo gallery.

The experience with calendars in mobile is not great, as they don't appear centered on the screen, and sometimes is not even possible to change months.
We need to fix this on the following calendars:

- Edit Blocked Period;
- Add New Blocked Period;

Also, we need to change the behaviour from the dropdown in the photo management (the units) to the native system.

## Screenshots

screen shot 2018-01-24 at 16 42 31
img_4174

### CODE REVIEW

**Review by AndreJoaquim — APPROVED** [review]

**Review by asantos00 — CHANGES_REQUESTED** [review]
It seems ok. 
The only thing I would ask you is to create a test where you use the uni-dropdown passing two different types of options, one of them being an array of strings, other being an array with key values. 
I think this is the way we can grant we're not breaking other platforms that are using the *old api* and that we will not break this in future just cause "it seemed ok to change this".

Tip 1: Pick a few world case uses of this in other platforms and use that to test ;) 
Tip 2: Use ember-responsive *mock screen size* helpers (https://github.com/freshbooks/ember-responsive#integration-tests) to grant you're testing this in desktop and mobile (since this is a mobile only bug)

**Comment by asantos00** [comment]
Your tests look great, great job! 🎖 Now I'm pretty much confident we're not actually breaking the API, and if we break, thanks to your tests, we will notice 👍

**Review by asantos00 — APPROVED** [review]
💯

**Review by telmaantunes — APPROVED** [review]
👍
